examples/buckling/CMakeLists.txt

```
@@ -4,36 +4,45 @@
 #
 # SPDX-License-Identifier: (BSD-3-Clause)
 
-if(TRIBOL_FOUND AND PETSC_FOUND)
-    blt_add_executable( NAME        buckling_cylinder
-                        SOURCES     cylinder.cpp
-                        OUTPUT_DIR ${EXAMPLE_OUTPUT_DIRECTORY}
-                        DEPENDS_ON  serac_physics
-                        )
+blt_add_executable( NAME        buckling_cylinder
+                    SOURCES     cylinder.cpp
+                    OUTPUT_DIR ${EXAMPLE_OUTPUT_DIRECTORY}
+                    DEPENDS_ON  serac_physics
+                    )
+
+if(SERAC_ENABLE_HIP)
+    blt_add_target_compile_flags(TO buckling_cylinder FLAGS "-fgpu-rdc")
+    blt_add_target_link_flags(TO buckling_cylinder FLAGS "-fgpu-rdc")
+endif()
+
+serac_add_example_test(NAME           buckling_cylinder_no_contact
+                       COMMAND        buckling_cylinder --no-contact
+                       NUM_MPI_TASKS  4)
+
+if(TRIBOL_FOUND)
+    blt_add_test(NAME           buckling_cylinder_fast
+                 COMMAND        buckling_cylinder --fast
+                 NUM_MPI_TASKS  4)
 
     serac_add_example_test(NAME           buckling_cylinder
                            COMMAND        buckling_cylinder
                            NUM_MPI_TASKS  4)
+endif()
 
-    # TODO Fix this test
-    # `InvIso: Newtons method did not converge.`
-    # configure_file(cylinder_petsc_options.yml
-    #                ${PROJECT_BINARY_DIR}/examples/buckling/cylinder_petsc_options.yml
-    #                COPYONLY)
-    #
-    # serac_add_example_test(NAME            buckling_cylinder_mortar_contact
-    #                        COMMAND         buckling_cylinder --contact --contact-type 1 --preconditioner 6 -options_file examples/buckling/cylinder_petsc_options.yml
-    #                        NUM_MPI_TASKS   4)
-
-    serac_add_example_test(NAME           buckling_cylinder_no_contact
-                           COMMAND        buckling_cylinder --no-contact
-                           NUM_MPI_TASKS  4)
-
-    install(
-        FILES
-            cylinder.cpp
-        DESTINATION
-            examples/serac/buckling
-    )
+if(TRIBOL_FOUND AND ENZYME_FOUND AND PETSC_FOUND)
+    configure_file(cylinder_petsc_options.yml
+                    ${PROJECT_BINARY_DIR}/examples/buckling/cylinder_petsc_options.yml
+                    COPYONLY)
 
+    serac_add_example_test(NAME            buckling_cylinder_mortar_contact
+                           COMMAND         buckling_cylinder --contact --contact-type 1 --preconditioner 6 -options_file examples/buckling/cylinder_petsc_options.yml
+                           NUM_MPI_TASKS   4
+                           WORKING_DIRECTORY ${PROJECT_BINARY_DIR})
```

```suggestion
                           COMMAND         buckling_cylinder --contact --contact-type 1 --preconditioner 6 -options_file ${PROJECT_BINARY_DIR}/examples/buckling/cylinder_petsc_options.yml
                           NUM_MPI_TASKS   4)
```

Didn't test this but I think it's cleaner assuming the issue was the path to the option file.